### DISCUSSION TITLE
Frequently running into "Maximum Processing Time"

### DISCUSSION THREAD

**Discussion by dhymik** [deepgram/community#585] [General help]
### Which Deepgram product are you using?

Deepgram API

### Details

I sent aac files (size approx. 256 MB, length approx. 120 minutes) of German audio for transcription through the API, using the .NET SDK. The processing time frequently reaches 10 minutes, when the request is aborted with `504: Gateway Timeout`, in accordance [with the docs](https://developers.deepgram.com/docs/getting-started-with-pre-recorded-audio#maximum-processing-time). 

For the request, I do not use any unusual options which are known to use up a lot of time. This is what I use:

```
 new PrerecordedTranscriptionOptions()
 {
     Model = "nova-2-general",
     Punctuate = true,
     SmartFormat = true,
     Language = "de",
     Keywords = "...some...",
     Diarize = true
 });

```

 Is there a way to get around this gateway timeout? Should this be taking so long? I do not mind the wait, however, when the request times out, I am left without a result. Sometimes, repeating the request works and I get a response with barely below 10 minutes.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jviso** [comment]
Hi @dhymik. This seems pretty unusual, so you could be discovering some interesting edge case in our system. Would you please share a few request IDs for 504s you've received? We'd like to look deeper at what's going on with your inputs.

**Reply by dhymik** [reply]
Hello @jviso I am very interested in you looking at it, since it frequently happens to me.

Here is a request ids of today: d9ce4a1a-82ab-4d10-9c63-230804285ea5

I tried several times this afternoon with the same file, and I always get a 503. Always, however, if I try at another time of day or another day, the same file goes throw, giving me the impression that it has to do with server load on your side.

I would be very gratefu if this can be resolved. It is really getting in my way often.

Kind regards! Mikael

**Reply by dhymik** [reply]
20 minutes later, the same file went through in 6 minutes 1 second with 200 and request id f3b7e9dc-c0bd-4217-a8ac-c3e7d157b1a0.
